src/git/repo.rs

```diff
@@ -965,6 +1077,134 @@ Either create it, or update the main branch setting by running:
         Ok(make_non_zero_oid(oid))
     }
+
+    /// Amends the provided parent commit in memory and returns the resulting tree.
+    ///
+    /// Only amends the files provided in the options, and only supports amending from
+    /// either the working tree or the index, but not both.
+    ///
+    /// See `Repo::cherry_pick_fast` for motivation for performing the operation
+    /// in-memory.
+    pub fn amend_fast(
+        &self,
+        parent_commit: &Commit,
+        opts: &AmendFastOptions,
+    ) -> eyre::Result<Tree> {
+        let parent_commit_pathbufs = self
+            .get_paths_touched_by_commit(parent_commit)?
+            .ok_or_else(|| {
+                eyre::eyre!(
+                    "Could not get paths touched by commit: {:?}",
+                    &parent_commit
+                )
+            })?
+            .into_iter()
+            .collect_vec();
+        let changed_paths: Vec<PathBuf> = {
+            let mut result: HashSet<PathBuf> = parent_commit_pathbufs.into_iter().collect();
+            match opts {
+                AmendFastOptions::FromIndex { paths } => result.extend(paths.iter().cloned()),
+                AmendFastOptions::FromWorkingTree { ref status_entries } => {
+                    for entry in status_entries {
+                        result.extend(entry.paths().iter().cloned());
+                    }
+                }
+            };
+            result.into_iter().collect_vec()
+        };
+        let changed_paths = changed_paths
+            .iter()
+            .map(|path| path.as_path())
+            .collect_vec();
+
+        let dehydrated_parent =
+            self.dehydrate_commit(parent_commit, changed_paths.as_slice(), true)?;
+        let dehydrated_parent_tree = dehydrated_parent.get_tree()?;
+
+        let repo_path = self
+            .get_working_copy_path()
+            .expect("repo working copy path");
+        let new_tree_entries: HashMap<PathBuf, Option<(NonZeroOid, i32)>> = match opts {
+            AmendFastOptions::FromWorkingTree { status_entries } => status_entries
+                .iter()
+                .map(|entry| {
+                    entry.paths().into_iter().map(move |path| {
+                        let file_path = &repo_path.join(&path);
+                        // Try to create a new blob OID based on the current on-disk
+                        // contents of the file in the working copy.
+                        match self.inner.blob_path(file_path) {
+                            Ok(oid) => Ok((
+                                path,
+                                Some((make_non_zero_oid(oid), entry.working_copy_file_mode.into())),
+                            )),
+                            // If the file doesn't exist, it needs to be explicitly marked as
+                            // such, as a tombstone to override if the file exists in the parent tree.
+                            Err(ref err) if err.code() == git2::ErrorCode::NotFound => {
```

Comment:
Why do we need to capture `err` by `ref` here?

Reply:
Whoops, I think this was a copy/paste fail.